modules/placement-driver-api/src/testFixtures/java/org/apache/ignite/internal/placementdriver/TestPlacementDriver.java

```diff
@@ -37,13 +41,21 @@
 @TestOnly
 public class TestPlacementDriver extends AbstractEventProducer<PrimaryReplicaEvent, PrimaryReplicaEventParameters>
         implements PlacementDriver {
-    private final TestReplicaMetaImpl primaryReplica;
+    private final Supplier<TestReplicaMetaImpl> primaryReplicaSupplier;
```

Comment:
Because, for example, in integration tests, the node ID is assigned only at the start of the node, and `TestPlacementDriver` is initialized before the start of the node.

Reply:
In that particular case ClusterService is started on line 125.

Reply:
This is not so, the `ClusterService#start` on line 166, you can try running it locally and see the result.
On line 125 the `ClusterService` is just created.

The second example `TestPlacementDriver` is created at `org/apache/ignite/internal/rebalance/ItRebalanceDistributedTest.java:817` and starts at `org.apache.ignite.internal.rebalance.ItRebalanceDistributedTest.Node#start`.

And there are several such places.

Reply:
@vldpyatkov Why we use different nodeNames (and nodeIds) for PlacementDriver in given test. Basically that means that we will have different primary replicas on different nodes, which is generally not expected. Is that correct?

Reply:
Ok, In that case it should be fixed in order to use same node.

Reply:
After a personal discussion, I fixed the tests and slightly changed the constructor in **TestPlacementDriver**.

Reply:
I also created a separate ticket IGNITE-20766 where a couple of integration tests will need to be fixed.